service/controller/v5/key/key.go

```diff
@@ -94,6 +98,10 @@ func BlobName(customObject providerv1alpha1.AzureConfig, role string) string {
 	return fmt.Sprintf("%s-%s-%s", VersionBundleVersion(customObject), cloudConfigVersion, role)
 }
 
+func CertificateEncryptionName(customObject providerv1alpha1.AzureConfig) string {
```

Comment:
```suggestion
func CertificateEncryptionSecretName(customObject providerv1alpha1.AzureConfig) string {
```

I think its clearer to have secret in the name.